src/SDKs/Azure.Base/data-plane/Azure.Base/Http/HttpPipelineOptions.cs

```diff
@@ -3,77 +3,138 @@
 
 using Azure.Base.Http.Pipeline;
 using System;
-using System.Buffers;
+using System.Collections.Generic;
 using System.ComponentModel;
-using System.Diagnostics;
-using System.Threading.Tasks;
 
 namespace Azure.Base.Http
 {
-    public partial struct HttpPipeline
+    public class HttpPipelineOptions
     {
-        public class Options
-        {
-            static readonly HttpPipelinePolicy s_default = new Default();
-
-            HttpPipelineTransport _transport;
+        ServiceContainer _container = new ServiceContainer();
 
-            public ArrayPool<byte> Pool { get; set; } = ArrayPool<byte>.Shared;
+        HttpPipelineTransport _transport;
+        List<HttpPipelinePolicy> _perCallPolicies;
+        List<HttpPipelinePolicy> _perRetryPolicies;
 
-            public HttpPipelineTransport Transport {
-                get => _transport;
-                set {
-                    if (value == null) throw new ArgumentNullException(nameof(value));
-                    _transport = value;
-                }
+        public HttpPipelineTransport Transport {
+            get => _transport;
+            set {
+                if (value == null) throw new ArgumentNullException(nameof(value));
+                _transport = value;
             }
+        }
 
-            public HttpPipelinePolicy TelemetryPolicy { get; set; } = s_default;
+        public HttpPipelinePolicy LoggingPolicy { get; set; }
 
-            public HttpPipelinePolicy LoggingPolicy { get; set; } = s_default;
+        public HttpPipelinePolicy RetryPolicy { get; set; }
 
-            public HttpPipelinePolicy RetryPolicy { get; set; } = s_default;
+        public bool DisableTelemetry { get; set; } = false;
```

Comment:
Could we change the name to `EnableTelemetry` ? Idk if it is just me, but I have to think a little more for the double negative everywhere to understand that we actually want telemetry so we are going to add a policy

Reply:
We want telemetry to be enabled by default, and we have a guideline that says that Boolean parameters and property values should by default be the detault(bool), i.e. false. Also, EnableTelemetry sounds (to me) like the user needs to do something (set the property) to enable telemetry, where in fact telemetry is enabled by default. 

BTW, @bartonjs, I know we have the guideline, but I cannot find it.

Reply:
Thanks for the clarification. I didn't know that.

Reply:
The best match I've found is

* DO provide sensible default values for all properties, ensuring that the defaults do not result in a security hole or terribly inefficient code.

For a complex struct it'd be "`default(T)` should represent valid defaults" (which I don't know is written down explicitly... the closest is that all-zero should be "valid").  For overloads we usually suggest that the new parameters have `default(T)` to mean "treat me like the smaller overload"

Since you're changing to a class you can use a the object initializer to default the field to true.

And, as bounced around the room, nothing says that a property can't return the negative of a storage value, e.g. the field is `_disableTelemetry` (default false) and the property inverts it.